lightning_training_studio/app/ui/src/components/ExperimentTable.tsx

```diff
@@ -118,10 +124,26 @@ function toProgress(experiment: ExperimentConfig) {
 }
 
 function runtimeTime(experiment: ExperimentConfig) {
-  if (experiment.end_time) {
+  if (experiment.end_time && experiment.progress) {
     return formatDurationStartEnd(Number(experiment.end_time), Number(experiment.start_time));
   }
-  return experiment.start_time ? String(formatDurationFrom(Number(experiment.start_time))) : <Box></Box>;
+  return experiment.start_time ? String(formatDurationFrom(Number(experiment.start_time))) : '';
+}
+
+function getTime() {
+  return Number(new Date().getTime() / 1000);
+}
+
+function timeLeft(experiment: ExperimentConfig) {
+  if (experiment.end_time || experiment.stage == 'stopped' || experiment.stage == 'failed') {
+    return '';
+  }
+  if (experiment.progress) {
+    const nowTime = getTime();
+    const estimatedEnd =
+      Number(experiment.start_time) + (100 * (nowTime - Number(experiment.start_time))) / experiment.progress;
```

Comment:
handle when progress is still 0 (and potentially elapsed time is also very small)?

Reply:
It won't happen. progress goes from null to defined.